### PR TITLE
[dv/clkmgr] Add stress testpoint as V3 milestone

### PR DESCRIPTION
**Pull request by matutem**
Require stress test development for V3 milestone.

Signed-off-by: Guillermo Maturana <maturana@google.com>

### CODE REVIEW

**Comment by sriyerg** [comment]
Please also import the testplan "hw/dv/tools/dvsim/testplans/stress_all_with_reset_testplan.hjson". 

@weicaiyang, that `stress_all_with_rand_reset` is marked as V2. IMO we can de-prioritize to V3, considering we have only found DV testbench issues and not real RTL issues. WDYT?

**Comment by sriyerg** [comment]
@matutem, looks like the automated intr and alert tests are now enabled. 
https://reports.opentitan.org/hw/ip/clkmgr/dv/latest/results.html

Please also uncomment those testplan imports at the top so that they get mapped properly?

**Comment by sriyerg** [comment]
> Please also import the testplan "hw/dv/tools/dvsim/testplans/stress_all_with_reset_testplan.hjson".
> 
> @weicaiyang, that `stress_all_with_rand_reset` is marked as V2. IMO we can de-prioritize to V3, considering we have only found DV testbench issues and not real RTL issues. WDYT?

Now that I think about it, the common stress testplan does not have a vanilla `stress` test - do you recall why?

**Comment by sriyerg** [comment]
@matute, would you also please add a testpoint that replicates the bug in:
https://github.com/lowRISC/opentitan/issues/6504

I think it should be possible to do that in clkmgr IP level and validate the fix (which IMO is not yet made).

**Comment by weicaiyang** [comment]
> > Please also import the testplan "hw/dv/tools/dvsim/testplans/stress_all_with_reset_testplan.hjson".
> > @weicaiyang, that `stress_all_with_rand_reset` is marked as V2. IMO we can de-prioritize to V3, considering we have only found DV testbench issues and not real RTL issues. WDYT?
> 
> Now that I think about it, the common stress testplan does not have the vanilla `stress` test - do you recall why?

we usually includes multiple sequences in stress_all, but how we run and what we include may vary. For example, most of blocks run sub-sequence sequentially, but xbar runs all sub-sequences in parallel. We may not include all sequences, for example, we don't include those timing sensitive tests that require zero_delays. It's good to document theses in testplan.

For stress_all_with_rand_reset, it's really just stress_all across with reset. DV users don't even need to create the sequence as it's created automatically like csr_test. Hence, we have common testplan for it.

**Comment by weicaiyang** [comment]
> Please also import the testplan "hw/dv/tools/dvsim/testplans/stress_all_with_reset_testplan.hjson".
> 
> @weicaiyang, that `stress_all_with_rand_reset` is marked as V2. IMO we can de-prioritize to V3, considering we have only found DV testbench issues and not real RTL issues. WDYT?

Agree, to change it to V3. Let me update that.

File: hw/ip/clkmgr/data/clkmgr_testplan.hjson [hunk 1]
```diff
@@ -153,7 +153,14 @@
       milestone: V2
       tests: []
     }
+    {
+      name: stress
+      desc: '''Create standard stress tests.'''
```
Comment (weicaiyang) [inline hw/ip/clkmgr/data/clkmgr_testplan.hjson:158]: nit: perhaps add a TODO to add what sequences we use in stress. It's not clear right now, but we should know after we have all the sub-sequences.

File: hw/ip/clkmgr/data/clkmgr_testplan.hjson [hunk 2]
```diff
@@ -153,7 +153,14 @@
...
+      milestone: V3
+      tests: []
```
Comment (weicaiyang) [inline hw/ip/clkmgr/data/clkmgr_testplan.hjson:160]:
```suggestion
      tests: [clkmgr_stress_all]
```

**Review by weicaiyang — APPROVED** [review]
LGTM with 2 nit

File: hw/ip/clkmgr/data/clkmgr_testplan.hjson (hunk 1 above)
Comment (matutem) [reply hw/ip/clkmgr/data/clkmgr_testplan.hjson:158]: Done.

File: hw/ip/clkmgr/data/clkmgr_testplan.hjson (hunk 2 above)
Comment (matutem) [reply hw/ip/clkmgr/data/clkmgr_testplan.hjson:160]: I thought we not should add the tests until they are runnable.

Comment (sriyerg) [reply hw/ip/clkmgr/data/clkmgr_testplan.hjson:186]: Agree with Matute here.

**Comment by sriyerg** [comment]
> > > Please also import the testplan "hw/dv/tools/dvsim/testplans/stress_all_with_reset_testplan.hjson".
> > > @weicaiyang, that `stress_all_with_rand_reset` is marked as V2. IMO we can de-prioritize to V3, considering we have only found DV testbench issues and not real RTL issues. WDYT?
> > 
> > 
> > Now that I think about it, the common stress testplan does not have the vanilla `stress` test - do you recall why?
> 
> we usually includes multiple sequences in stress_all, but how we run and what we include may vary. For example, most of blocks run sub-sequence sequentially, but xbar runs all sub-sequences in parallel. We may not include all sequences, for example, we don't include those timing sensitive tests that require zero_delays. It's good to document theses in testplan.
> 
> For stress_all_with_rand_reset, it's really just stress_all across with reset. DV users don't even need to create the sequence as it's created automatically like csr_test. Hence, we have common testplan for it.

I see, thanks for the explanation. It is good to allow users to write their own description for the stress test.

File: hw/ip/clkmgr/data/clkmgr_testplan.hjson (hunk 2 above)
Comment (weicaiyang) [reply hw/ip/clkmgr/data/clkmgr_testplan.hjson:186]: I see. Thanks for correcting me.

**Review by sriyerg — APPROVED** [review]

**Pull request merged**